core/embed/rust/src/ui/layout/confirm_t1.rs

```diff
@@ -0,0 +1,146 @@
+use core::convert::{TryFrom, TryInto};
+
+use crate::{
+    error::Error,
+    micropython::{buffer::Buffer, func::unpack_args, obj::Obj},
+    ui::{
+        component::{
+            model_t1::{theme, Button, ButtonPos, Dialog, DialogMsg},
+            Child, Text,
+        },
+        display,
+    },
+    util,
+};
+
+use super::obj::LayoutObj;
+
+impl<T> TryFrom<DialogMsg<T>> for Obj
+where
+    Obj: TryFrom<T>,
+    Error: From<<T as TryInto<Obj>>::Error>,
+{
+    type Error = Error;
+
+    fn try_from(val: DialogMsg<T>) -> Result<Self, Self::Error> {
+        match val {
+            DialogMsg::Content(c) => Ok(c.try_into()?),
+            DialogMsg::LeftClicked => 1.try_into(),
+            DialogMsg::RightClicked => 2.try_into(),
+        }
+    }
+}
+
+#[no_mangle]
+extern "C" fn ui_layout_new_confirm_action(n_args: usize, args: *const Obj) -> Obj {
```

Comment:
Could we use kwargs here?

Reply:
Good idea, I converted this to kwargs in ef1b60347a3701ee3e2a1c85d6255fa3ef7c98d2 and c7b68d269d9641146d843a7d2035a1ffdf500edc.